tools/run-fbp-tests

```diff
@@ -76,137 +174,130 @@ def collect_tests(tests):
-                    print()
-                    continue
-            elif out != expected_out:
-                failed += 1
-                print("FAILED running %s: wrong output" % t)
-                for line in difflib.unified_diff(out.splitlines(True), expected_out.splitlines(True)):
-                    sys.stdout.write(line)
-                print()
-                continue
-        if expected_code != code:
-            failed += 1
-            print("FAILED running %s: wrong exit code, got %d but expected %d" % (t, code, expected_code))
-            print("Output:")
-            print(out)
-            print()
+        curr = FbpTest(t)
+        if curr.is_skip(skips):
+            print("SKIPPED running %s: %s" % (t, curr.skip_msg))
+            status["skipped"] += 1
             continue
-        if expected_out is None and len(out) > 0:
-            print("IGNORING unexpected output from", t)
-            print(out)
-        passed += 1
 
-    print("%d tests passed." % passed)
-    if skipped > 0:
-        print("%d tests skipped." % skipped)
-    if failed > 0:
-        print("%d tests failed." % failed)
+        if not args.compiled:
+            cmd = cmd_prefix + [runner, path["file"]]
+            test_cwd = path["dir"]
+        else:
+            exe = "%s/%s" % (args.compiled_dir, path["file"].replace(".fbp", ""))
+            cmd = cmd_prefix + [exe]
+            test_cwd = "./"
+
+        out, code = sh(cmd, cwd=test_cwd)
+        result = curr.check_result(out, code)
+        status[result] += 1
+        if result == "failed":
+            failures = True
+
+    for k,v in status.items():
+        if v > 0:
+            print("%d tests %s." % (v, k))
+
+    if failures:
         sys.exit(1)
+
+def get_skip_list(args):
+    result = []
+    for curr in args.skip:
+        skip = getattr(FbpTest, curr.upper(), None)
+        if isinstance(skip, int):
+            result.append(skip)
+        else:
+            print("Unknown skip spec: %s" % curr)
+    return result
+
+def query_tests(args):
+    result = []
+    skips = get_skip_list(args)
+    tests = collect_tests(args.tests)
+    for t in sorted(set(tests)):
+        curr = FbpTest(t)
+        if curr.is_skip(skips):
+            continue
+        result.append(t)
+    print(" ".join(result))
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--runner", help="Path to the sol-fbp-runner program",
+                        default="build/soletta_sysroot/usr/bin/sol-fbp-runner", type=str)
+    parser.add_argument("--valgrind", help="Enable valgrind, if path not passed, assume "
+                        "default location in the system", nargs='?', const="/usr/bin/valgrind",
+                        type=str)
+    parser.add_argument("--valgrind-supp", help="Path to valgrind's suppression file",
+                        type=str)
+    parser.add_argument("--verbose", help="Be verbose", dest="verbose", action="store_true")
```

Comment:
I like the fact that the help for the option that enables verbosity is quite terse. :)